src/wallet/rpcdump.cpp

```diff
@@ -660,19 +660,32 @@ UniValue dumpwallet(const JSONRPCRequest& request)
         std::string strAddr = CBitcoinAddress(keyid).ToString();
         CKey key;
         if (pwallet->GetKey(keyid, key)) {
+            CKeyMetadata meta;
+            {
+                const auto meta_it = pwallet->mapKeyMetadata.find(keyid);
+                if (meta_it != pwallet->mapKeyMetadata.end()) {
+                    meta = meta_it->second;
+                }
+            }
             file << strprintf("%s %s ", CBitcoinSecret(key).ToString(), strTime);
             if (pwallet->mapAddressBook.count(keyid)) {
                 file << strprintf("label=%s", EncodeDumpString(pwallet->mapAddressBook[keyid].name));
             } else if (keyid == masterKeyID) {
                 file << "hdmaster=1";
             } else if (mapKeyPool.count(keyid)) {
                 file << "reserve=1";
-            } else if (pwallet->mapKeyMetadata[keyid].hdKeypath == "m") {
+            } else if (meta.hdKeypath == "m") {
                 file << "inactivehdmaster=1";
             } else {
                 file << "change=1";
             }
-            file << strprintf(" # addr=%s%s\n", strAddr, (pwallet->mapKeyMetadata[keyid].hdKeypath.size() > 0 ? " hdkeypath="+pwallet->mapKeyMetadata[keyid].hdKeypath : ""));
+            if (!meta.hdKeypath.empty()) {
+                file << " hdkeypath=" + meta.hdKeypath;
```

Comment:
What is the reason to remove this out of the comment section?

Reply:
It's metadata that should be restored. (our `importwallet` doesn't support it, but that's another matter)